benchmarks/span/scenario.py

```diff
@@ -0,0 +1,99 @@
+import pyperf
+from util import gen_metrics
+from util import gen_tags
+
+from ddtrace import Span
+
+
+SPAN_START_VARIANTS = [dict(nspans=10000), dict(nspans=10000, trace_id=1)]
+
+
+def time_start_span(loops):
+    range_it = range(loops)
+    t0 = pyperf.perf_counter()
+    for _ in range_it:
+        gen_traces(**variant)
+    dt = pyperf.perf_counter() - t0
+    return dt
+
+
+def gen_traces(nspans=1, trace_id=None):
+    for _ in range(0, nspans):
+        Span(None, "test.op", resource="resource", service="service", trace_id=trace_id)
+
+def gen_spans(nspans=1, trace_id=None):
+    spans = []
+    for _ in range(0, nspans):
+        spans.append(Span(None, "test.op", resource="resource", service="service", trace_id=trace_id))
+    return spans
+
+
+def add_tags(tags, span):
+    for i in range(len(tags[0])):
+        span.set_tag(tags[0][i], tags[1][i])
+
+
+def add_metrics(metrics, span):
+    for i in range(len(metrics[0])):
+        span.set_tag(metrics[0][i], metrics[1][i])
+
+def finish_spans(spans):
+    for span in spans:
+        span.finish()
+
+
+def time_add_tags(loops, span, tags):
+    range_it = range(loops)
+    t0 = pyperf.perf_counter()
+    for _ in range_it:
+        add_tags(span, tags)
+    dt = pyperf.perf_counter() - t0
+    return dt
+
+
+def time_add_metrics(loops, span, metrics):
+    range_it = range(loops)
+    t0 = pyperf.perf_counter()
+    for _ in range_it:
+        add_metrics(span, metrics)
+    dt = pyperf.perf_counter() - t0
+    return dt
+
+def time_finish_span(loops, spans):
+    range_it = range(loops)
+    t0 = pyperf.perf_counter()
+    for _ in range_it:
+        finish_spans(spans)
+    dt = pyperf.perf_counter() - t0
+    return dt
+
+
+if __name__ == "__main__":
+    runner = pyperf.Runner()
+    runner.metadata["scenario"] = "start_span"
+    for variant in SPAN_START_VARIANTS:
+        name = "|".join(f"{k}:{v}" for (k, v) in variant.items())
+        metadata = {}
+        runner.bench_time_func(name, time_start_span, metadata=metadata)
+
+    for variant in [dict(ntags=500, ltags=100), dict(ntags=1000, ltags=300)]:
+        span = Span(None, "test.op", resource="resource", service="service", trace_id=1)
+        tags = gen_tags(**variant)
+        name = "|".join(f"{k}:{v}" for (k, v) in variant.items())
+        metadata = {}
+        runner.bench_time_func(name, time_add_tags, tags, span, metadata=metadata)
+
+    for variant in [dict(nmetrics=500, lmetrics=100), dict(nmetrics=1000, lmetrics=300)]:
+        span = Span(None, "test.op", resource="resource", service="service", trace_id=1)
+        metrics = gen_metrics(**variant)
+        name = "|".join(f"{k}:{v}" for (k, v) in variant.items())
+        metadata = {}
+        runner.bench_time_func(name, time_add_metrics, metrics, span, metadata=metadata)
```

Comment:
make sure to set the name to include a distinction for the specific test. 

https://github.com/DataDog/dd-trace-py/pull/2646/files#diff-d45ee81b3bd9d274cead35e0e122aad0185f235fd52ce72c4e019b81a2bba2a2R96